ngi_reports/utils/entities.py

```diff
@@ -43,18 +57,237 @@ def __init__(self):
+
         self.lanes = OrderedDict()
-        self.name = ""
-        self.run_name = ""
-        self.run_setup = []
-        self.seq_meth = ""
-        self.type = ""
-        self.run_params = {}
-        self.chemistry = {}
-        self.casava = None
-        self.seq_software = {}
+        self.fc_sample_qvalues = defaultdict(dict)
+
+    def populate_illumina_flowcell(self, log, **kwargs):
+        fc_instrument = self.fc_details.get("RunInfo", {}).get("Instrument", "")
+        fc_runparameters = self.fc_details.get("RunParameters", {})
+        self.run_setup = self.fc_details.get("RunInfo").get("Reads")
+        self.sample_sheet_data = self.fc_details.get("samplesheet_csv")
+
+        if "-" in self.name:
+            self.type = "MiSeq"
+            self.chemistry = {
+                "Chemistry": fc_runparameters.get(
+                    "ReagentKitVersion", fc_runparameters.get("Sbs")
+                )
+            }
+            self.seq_software = {
+                "RTAVersion": fc_runparameters.get("RTAVersion"),
+                "ApplicationVersion": fc_runparameters.get("MCSVersion"),
+            }
+
+        elif fc_instrument.startswith("A"):
+            self.type = "NovaSeq6000"
+            self.chemistry = {
+                "WorkflowType": fc_runparameters.get("WorkflowType"),
+                "FlowCellMode": fc_runparameters.get("RfidsInfo", {}).get(
+                    "FlowCellMode"
+                ),
+            }
+            self.seq_software = {
+                "RTAVersion": fc_runparameters.get(
+                    "RTAVersion", fc_runparameters.get("RtaVersion")
+                ),
+                "ApplicationName": fc_runparameters.get(
+                    "ApplicationName", fc_runparameters.get("Application")
+                ),
+                "ApplicationVersion": fc_runparameters.get("ApplicationVersion"),
+            }
+
+        elif fc_instrument.startswith("LH"):
+            self.type = "NovaSeqXPlus"
+            self.chemistry = {"RecipeName": fc_runparameters.get("RecipeName")}
+            self.seq_software = {
+                "ApplicationName": fc_runparameters.get("Application"),
+                "ApplicationVersion": fc_runparameters.get("SystemSuiteVersion"),
+            }
+
+        elif fc_instrument.startswith("VH"):
+            self.type = "NextSeq2000"
+            NS2000_FC_PAT = re.compile("P[1,2,3]")
+            self.chemistry = {
+                "Chemistry": NS2000_FC_PAT.findall(
+                    fc_runparameters.get("FlowCellMode")
+                )[0]
+            }
+            self.seq_software = {
+                "RTAVersion": fc_runparameters.get("RtaVersion"),
+                "ApplicationName": fc_runparameters.get("ApplicationName"),
+                "ApplicationVersion": fc_runparameters.get("ApplicationVersion"),
+            }
+
+        else:
+            log.warning("Unknown sequencing instrument detected: {fc_instrument}")
```

Comment:
Should it break execution here, do you think? Would it have the rest of the fields below?